src/cfb.rs

```diff
@@ -78,7 +80,9 @@ impl Cfb {
     pub fn new<R: Read>(mut reader: &mut R, len: usize) -> Result<Cfb, CfbError> {
         // load header
         let (h, mut difat) = Header::from_reader(&mut reader)?;
-        let mut sectors = Sectors::new(h.sector_size, Vec::with_capacity(len));
+        assert_ne!(h.sector_size, 0);
+        let maxid = len / h.sector_size;
+        let mut sectors = Sectors::new(h.sector_size, Vec::with_capacity(len), maxid as u32);
```

Comment:
Wouldn't it be simpler to just use a fixed size for capacity like `Vec::with_capacity(1024)` here and elsewhere. I doubt that these are cases where over or under allocation of Vec will make any difference to performance.